src/Avalonia.Controls/Presenters/ScrollContentPresenter.cs

```diff
@@ -1065,5 +1065,14 @@ private static (double previous, double next) FindNearestSnapPoint(IReadOnlyList
 
             return snapPointsInfo;
         }
+
+        private Vector CheckFlowDirection(Vector delta)
```

Comment:
Nit: consider removing `Check` from the name, which usually implies a boolean result. (e.g. `AdjustDeltaForFlowDirection`, `GetDirectionDelta` or something better - as always, naming is hard!).